### PR TITLE
Eswe 937 patch education employment

### PR DESCRIPTION
**Pull request by sacintha**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,82 +1,95 @@
...
-tasks.jacocoTestReport {
-  dependsOn(tasks.test)
-  reports {
-    html.required.set(true)
-    xml.required.set(true)
+    finalizedBy(named("jacocoTestReport"))
+    named("assemble") {
+      doFirst {
+        delete(
+          fileTree(project.layout.buildDirectory.get())
```
Comment (andrewrlee) [inline build.gradle.kts:89]: I don't think this is needed? - we don't do this in other projects

Comment (andrewrlee) [reply build.gradle.kts:89]: (something similar was previously incorrectly deleting the main app jar preventing deployments)

Comment (andrewrlee) [reply build.gradle.kts:89]: (something similar was previously incorrectly deleting the main app jar preventing deployments)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationemployment/api/config/ApplicationInsightsConfiguration.kt [hunk 2]
```diff
@@ -15,7 +15,7 @@ class ApplicationInsightsConfiguration {
   @Bean
   @ConditionalOnExpression("T(org.apache.commons.lang3.StringUtils).isBlank('\${applicationinsights.connection.string:}')")
   fun telemetryClient(): TelemetryClient {
-    log.warn("Application insights configuration missing, returning dummy bean instead")
+    uk.gov.justice.digital.hmpps.educationemployment.api.config.ApplicationInsightsConfiguration.Companion.log.warn("Application insights configuration missing, returning dummy bean instead")
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/educationemployment/api/config/ApplicationInsightsConfiguration.kt:18]: guess this whole package ref isn't required

File: src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationemployment/api/integration/util/TestData.kt [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+      modifiedTime,
+      createdBy,
+      modifiedTime,
+      "1.0",
+      CapturedSpringConfigValues.objectMapper.readTree(CapturedSpringConfigValues.objectMapper.writeValueAsString(profile)),
+      CapturedSpringConfigValues.objectMapper.readTree("{}"),
+      booleanTrue,
+    )
+
+    val updatedReadinessProfile = Optional.of(
```
Comment (andrewrlee) [inline src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationemployment/api/integration/util/TestData.kt:277]: not necessary, in kotlin you wouldn't use Optional. 
Would just use nullable values and things like `?.let{}` to do `map`/`flatmap`

File: build.gradle.kts [hunk 4]
```diff
@@ -8,75 +7,80 @@ plugins {
...
-  implementation("org.springframework.boot:spring-boot-starter-security")
-  implementation("org.springframework.boot:spring-boot-starter-cache")
-  implementation("uk.gov.justice.service.hmpps:hmpps-sqs-spring-boot-starter:3.0.0")
-  implementation("io.hypersistence:hypersistence-utils-hibernate-63:3.7.5")
-  implementation("io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations:2.4.0")
-  implementation("com.microsoft.azure:applicationinsights-logging-logback:2.6.4")
-
-  developmentOnly("org.springframework.boot:spring-boot-devtools")
-
+  implementation("org.springframework.boot:spring-boot-starter-hateoas")
```
Comment (andrewrlee) [inline build.gradle.kts:13]: surprised this is used

File: build.gradle.kts [hunk 5]
```diff
@@ -8,75 +7,80 @@ plugins {
...
-  implementation("org.apache.commons:commons-lang3:3.14.0")
-  implementation("org.ehcache:ehcache:3.10.8")
-  implementation("com.zaxxer:HikariCP:5.1.0")
-  implementation("org.hibernate.orm:hibernate-community-dialects")
-
-  // Database dependencies
   runtimeOnly("org.flywaydb:flyway-database-postgresql")
   runtimeOnly("org.postgresql:postgresql:42.7.3")
-
+  implementation("org.testcontainers:localstack")
```
Comment (andrewrlee) [inline build.gradle.kts:22]: these should be `testimplementation`?

File: build.gradle.kts [hunk 6]
```diff
@@ -8,75 +7,80 @@ plugins {
...
-  implementation("io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations:2.4.0")
-  implementation("com.microsoft.azure:applicationinsights-logging-logback:2.6.4")
-
-  developmentOnly("org.springframework.boot:spring-boot-devtools")
-
+  implementation("org.springframework.boot:spring-boot-starter-hateoas")
   implementation("org.springframework.boot:spring-boot-starter-oauth2-resource-server")
+  implementation("org.springdoc:springdoc-openapi-starter-webmvc-ui:2.3.0")
+  implementation("com.fasterxml.jackson.core:jackson-databind")
+  implementation("io.jsonwebtoken:jjwt-impl:0.12.5")
```
Comment (andrewrlee) [inline build.gradle.kts:17]: these are listed a `testImplementation` too?  do they needed to be listed only here instead or can you get away with them just being `testImplementation`?

Comment (sacintha) [reply build.gradle.kts:17]: we need this here as well

**Review by andrewrlee — DISMISSED** [review]
LGTM

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**